### ISSUE TITLE
parseImports does not strip CSS comments, so an `@import "tailwindcss"` written inside a comment marks a partial theme as a full entry

### ISSUE THREAD

**Issue by kang8** [shadcn-ui/lint#36]
## Summary

Follow-up to #13 / #31. On 0.1.1 the fallback added in #31 still never fires in my project and nothing is printed — but the cause sits upstream of `themeFileFor`.

In `src/project/theme.ts`, `readTheme` calls `parseImports(css)` on the **raw** source, while `parseClassSelectors` in the same file strips `/* … */` first. My `theme.css` documents the consumer contract in a block comment that happens to spell the import out, and the regex

```js
/@import\s+(?:url\(\s*)?["']([^"']+)["']\s*\)?[^;]*;/g
```

matches that comment text — `[^;]*;` runs past the end of the comment to the next `;` anywhere in the file. So `read.tailwind` becomes `true`, `tailwindEntryFor` returns the partial unchanged, and both the discovery fallback and the `…does not import Tailwind` warning are skipped. `no-unknown-classes` then validates against a theme with no base utilities: 308 ordinary classes (`p-4`, `text-sm`, …) reported unknown, with nothing on stderr.

The silence is the expensive part — the plugin believes the setup is correct, so there is no thread to pull on.

## Reproduction

`components.json`

```json
{ "$schema": "https://ui.shadcn.com/schema.json", "tailwind": { "css": "src/theme.css" } }
```

`src/theme.css` — a token-only partial, with a comment that mentions the import

```css
/* Consumers own the `@import 'tailwindcss'` entry point and must import this
 * file alongside it. */
@theme inline {
  --color-brand: #123456;
}
```

`src/app.css` — the real entry

```css
@import 'tailwindcss';
@source './';
@import './theme.css';
```

`src/page.tsx`

```tsx
export default function Page() {
  return <div className="flex p-4 text-sm bg-brand" />
}
```

With `no-unknown-classes: error`:

- **As written:** `p-4` and `text-sm` are reported unknown. No warning.
- **Reword the comment so it contains no `@import`:** the expected `components.json sets tailwind.css to src/theme.css, which does not import Tailwind…` warning appears and the run is clean.

In my repo that one-line comment edit took `no-unknown-classes` from 308 errors to 0 — nothing else changed.

## Suggested fix

Strip comments in `parseImports` the way `parseClassSelectors` already does. #6 is the same file's comment handling from the other direction (a stripper that eats too much, because it does not know about string literals), so a shared string-aware helper would settle both.

## Secondary: discovery cannot see a dot-directory

After rewording the comment, the fallback still found no entry in that package: `cssFilesUnder` skips every entry whose name starts with `.`, and the only stylesheet importing Tailwind inside the package is `.storybook/preview.css` — where Storybook conventionally puts it. `no-unknown-classes` therefore runs on the bundled grammar and prints its warning on every lint run. Not skipping dot-directories (or at least not `.storybook`) would let discovery land on the real entry.

## Environment

`@shadcn/lint` 0.1.1, eslint 10.10.0, tailwindcss 4.3.3, Node 26.8.2, pnpm workspace (monorepo, `components.json` inside the component package).
